src/hayhooks/cli/mcp.py

```diff
@@ -28,6 +27,12 @@ def run(
     additional_python_path: Annotated[
         Optional[str], typer.Option(help="Additional Python path to add to sys.path")
     ] = settings.additional_python_path,
+    json_response: Annotated[
+        bool, typer.Option("--json-response", "-j", help="Enable JSON responses instead of SSE streams")
```

Comment:
What's this switch for? How would I as a user decide to have it true/false? And I thought I understood mcp 😕

Reply:
Hmm yeah this might be a bit obscure!

Basically [if you enable this](https://github.com/modelcontextprotocol/python-sdk/blob/1bb16fa96ebb7440f481c47798f2e51878b78bae/src/mcp/server/streamable_http.py#L384), the `StreamableHTTPServerTransport` response will be a JSON response of type `application/json`, so basically the JSON-RPC response will be returned as-is (no stream).

If false, `StreamableHTTPServerTransport` response will be a regular SSE Stream of type `text/event-stream`.

For the end user it shouldn't be important, since clients will always supports SSE (via StreamableHTTP). It's a quite technical option I know, I've used it in tests to check that `/mcp` endpoint works.